paimon-common/src/main/java/org/apache/paimon/data/variant/VariantPathSegment.java

```diff
@@ -99,6 +70,36 @@ public static PathSegment[] parse(String str) {
             throw new IllegalArgumentException("Invalid path: " + str);
         }
 
-        return segments.toArray(new PathSegment[0]);
+        return segments.toArray(new VariantPathSegment[0]);
+    }
+
+    /** A path segment for object extraction. */
+    public static class ObjectExtraction extends VariantPathSegment {
```

Comment:
The parent abstract class is all static, and this class needs not inherit anything from the parent class. Not sure why this inheritance is applied.